### ISSUE TITLE
Issue when transforming WeakSegmentsPerformance() result to json

### ISSUE THREAD

**Issue by nuriaamenabar-vf** [deepchecks/deepchecks#2804]
When running WeakSegmentsPerformance().run(...).to_json(), the result's value field — which contains both weak_segments (a DataFrame) and avg_score — is passed to the JSON serializer as a dictionary.
This causes the output to be flattened instead of remaining a readable dictionary-like object.

For example, in all intermediate steps before calling .to_json(), the weak_segments are still a readable nested object.
However, after calling .to_json(), the structure is lost and the output looks like this:

<img width="2545" height="392" alt="Image" src="https://github.com/user-attachments/assets/db6d9378-4e10-4fce-820a-d0793b12ff15" />

Instead of a readable dictionary-like object.

```
from deepchecks.tabular.checks.model_evaluation import WeakSegmentsPerformance
from deepchecks.tabular.datasets.classification.lending_club import (
    load_data, load_fitted_model)
from deepchecks.tabular.checks import WeakSegmentsPerformance
from sklearn.metrics import make_scorer, f1_score

_, test_ds = load_data()
model = load_fitted_model()



scorer = {'f1': make_scorer(f1_score, average='micro')}
check = WeakSegmentsPerformance()
result = check.run(test_ds, model)
print(result) #Readable dict
print(result.to_json()) #Not readable
```

Package Version: 0.18;Page: api/deepchecks.tabular

**Comment by Solaris-star** [comment]
Taking this: will make `normalize_value` recursively convert nested DataFrames to record lists so `WeakSegmentsPerformance().run(...).to_json()` keeps a readable nested structure.
